samples/embedded/main.c

```diff
@@ -0,0 +1,215 @@
+      IREE_HAL_ELEMENT_TYPE_FLOAT_32, IREE_HAL_ENCODING_TYPE_DENSE_ROW_MAJOR,
+      (iree_hal_buffer_params_t){
+          .usage = IREE_HAL_BUFFER_USAGE_DEFAULT,
+          .type = IREE_HAL_MEMORY_TYPE_DEVICE_LOCAL,
+      },
+      iree_make_const_byte_span(X, sizeof(X)), &arg2));
+  IREE_RETURN_IF_ERROR(iree_hal_buffer_view_allocate_buffer(
+      iree_runtime_session_device_allocator(session), IREE_ARRAYSIZE(shape_y), shape_y,
+      IREE_HAL_ELEMENT_TYPE_FLOAT_32, IREE_HAL_ENCODING_TYPE_DENSE_ROW_MAJOR,
+      (iree_hal_buffer_params_t){
+          .usage = IREE_HAL_BUFFER_USAGE_DEFAULT,
+          .type = IREE_HAL_MEMORY_TYPE_DEVICE_LOCAL,
+      },
+      iree_make_const_byte_span(y, sizeof(y)), &arg3));
+
+  // Setup call inputs with our buffers.
+  IREE_RETURN_IF_ERROR(
+    iree_runtime_call_inputs_push_back_buffer_view(&call, arg0));
+  iree_hal_buffer_view_release(arg0);
+  IREE_RETURN_IF_ERROR(
+    iree_runtime_call_inputs_push_back_buffer_view(&call, arg1));
+  iree_hal_buffer_view_release(arg1);
+  IREE_RETURN_IF_ERROR(
+    iree_runtime_call_inputs_push_back_buffer_view(&call, arg2));
+  iree_hal_buffer_view_release(arg2);
+  IREE_RETURN_IF_ERROR(
+    iree_runtime_call_inputs_push_back_buffer_view(&call, arg3));
+  iree_hal_buffer_view_release(arg3);
+
+  // Invoke the function
+  IREE_RETURN_IF_ERROR(iree_runtime_call_invoke(&call, /*flags=*/0));
+
+  // Update weights
+  iree_hal_buffer_view_t* result = NULL;
+  IREE_RETURN_IF_ERROR(iree_runtime_call_outputs_pop_front_buffer_view(
+    &call, &result));
+  IREE_RETURN_IF_ERROR(iree_hal_device_transfer_d2h(
+    iree_runtime_session_device(session),
+    iree_hal_buffer_view_buffer(result), 0, &w,
+    sizeof(w), IREE_HAL_TRANSFER_BUFFER_FLAG_DEFAULT,
+    iree_infinite_timeout()));
+
+  // Update bias
+  IREE_RETURN_IF_ERROR(iree_runtime_call_outputs_pop_front_buffer_view(
+    &call, &result));
+  IREE_RETURN_IF_ERROR(iree_hal_device_transfer_d2h(
+    iree_runtime_session_device(session),
+    iree_hal_buffer_view_buffer(result), 0, &b,
+    sizeof(b), IREE_HAL_TRANSFER_BUFFER_FLAG_DEFAULT,
+    iree_infinite_timeout()));
+
+  // Update loss
+  IREE_RETURN_IF_ERROR(iree_runtime_call_outputs_pop_front_buffer_view(
+    &call, &result));
+  IREE_RETURN_IF_ERROR(iree_hal_device_transfer_d2h(
+    iree_runtime_session_device(session),
+    iree_hal_buffer_view_buffer(result), 0, &loss,
+    sizeof(loss), IREE_HAL_TRANSFER_BUFFER_FLAG_DEFAULT,
+    iree_infinite_timeout()));
+  iree_hal_buffer_view_release(result);
+
+  return iree_ok_status();
+}
+
+iree_status_t run_sample(iree_string_view_t bytecode_module_path,
+                         iree_string_view_t driver_name) {
+  iree_status_t status = iree_ok_status();
+
+  //===-------------------------------------------------------------------===//
+  // Instance configuration (this should be shared across sessions).
+  fprintf(stdout, "Configuring IREE runtime instance and '%s' device\n",
+          driver_name.data);
```

Comment:
nit: consistently use `fprintf(stdout,` or `printf(` in the file

Reply:
Done